src/textbuf.cpp

```diff
@@ -388,7 +385,7 @@ bool Textbuf::MovePos(uint16_t keycode)
 			return true;
 
 		case WKC_END:
-			this->caretpos = this->bytes - 1;
+			this->caretpos = static_cast<uint16_t>(this->buf.size());
```

Comment:
do I dare to ask if we shouldn't upgrade `caretpos` to a `size_t` ?

Reply:
You dared, but... if I were to change `caretpos`, it would stick out in the `Textbuf` `struct`, e.g. why `caret_pos` but not `max_chars` or `max_bytes`? Then 10 fields get changed to `size_t`, and quite a few parameters. That really feels like separate PR territory to me.